pxr/usdImaging/usdImaging/lightAdapter.cpp

```diff
@@ -80,6 +80,42 @@ UsdImagingLightAdapter::_RemovePrim(SdfPath const& cachePath,
                                     UsdImagingIndexProxy* index)
 {
     index->RemoveSprim(HdPrimTypeTokens->light, cachePath);
+    UsdImaging_CollectionCache &collectionCache = _GetCollectionCache();
+    SdfPath lightLinkPath = cachePath.AppendProperty(UsdImagingTokens->collectionLightLink);
+    collectionCache.RemoveCollection(GetDelegate()->GetStage(), lightLinkPath);
+    SdfPath shadowLinkPath = cachePath.AppendProperty(UsdImagingTokens->collectionShadowLink);
+    collectionCache.RemoveCollection(GetDelegate()->GetStage(), shadowLinkPath);
+}
+
+bool
+UsdImagingLightAdapter::_UpdateCollectionsChanged(UsdPrim const& prim, SdfPath const& cachePath) const
+{
+    UsdImaging_CollectionCache &collectionCache = _GetCollectionCache();
+    auto getCollectionHash = [&collectionCache] (const UsdCollectionAPI& api) -> size_t {
+        const TfToken id = collectionCache.UpdateCollection(api);
+        const UsdImaging_CollectionCache::Query* query = nullptr;
+        collectionCache.GetMembershipQuery(id, &query);
+        return query != nullptr ? query->GetHash() : 0;
+    };
+    UsdLuxLight light(prim);
+    const size_t newLightCollectionHash = getCollectionHash(light.GetLightLinkCollectionAPI());
+    const size_t newShadowCollectionHash = getCollectionHash(light.GetShadowLinkCollectionAPI());
```

Comment:
We might have gone over this already, but why are we hashing the collections?  Can we not have "UpdateCollection" tell us whether the membership has changed?

Reply:
Indeed! Would this signature be OK?
```
TfToken
UsdImaging_CollectionCache::UpdateCollection(UsdCollectionAPI const& c, bool* changed)
```

Reply:
Actually I'll just return a bool instead of the TfToken as the return value isn't needed anymore...